exchange/src/wrapper/algo_wrapper/binary/binary_wrapper.cpp

```diff
@@ -0,0 +1,143 @@
+#include "binary_wrapper.hpp"
+
+#include "common/util.hpp"
+#include "wrapper/messaging/exchange_communicator.hpp"
+
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/select.h>
+#include <unistd.h>
+
+#include <cerrno>
+#include <cstdlib>
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <stdexcept>
+
+namespace {
+namespace fs = std::filesystem;
+
+std::pair<int, std::filesystem::path>
+get_temp_file()
+{
+#ifdef __APPLE__
+    std::string template_path = (fs::temp_directory_path() / "algoXXXXXX").string();
```

Comment:
This is unclear. Might be a skill issue on my end but I feel like this could be done differently